### DISCUSSION TITLE
Unable to reproduce aluminum bulk modulus

### DISCUSSION THREAD

**Discussion by shivang-22** [CederGroupHub/chgnet#84] [Q&A] [answered]
Hello all, I am trying to calculate the bulk modulus of aluminum using the code below. I used the DeformedStructureSet technique to create different strains on the material, and then using CHGNet predict the stresses corresponding to each strain. The aluminum cif file was taken directly from the Materials Project website.

```
from chgnet.model.model import CHGNet
from pymatgen.core import Structure
from pymatgen.analysis.elasticity.strain import Deformation, DeformedStructureSet, Strain
from pymatgen.analysis.elasticity.stress import Stress
from pymatgen.analysis.elasticity import ElasticTensor

chgnet = CHGNet.load()
structure = Structure.from_file('cif_files/Al.cif')
prediction = chgnet.predict_structure(structure)

eq_stress = Stress(prediction['s'])

deformed_structure = DeformedStructureSet(structure)

strain_list = [i.green_lagrange_strain for i in deformed_structure.deformations]
stress_list = []

for i in range(len(deformed_structure)):
    prediction = chgnet.predict_structure(deformed_structure[i])
    stress_list.append(Stress(prediction['s']))

elastic_tensor = ElasticTensor.from_independent_strains(strain_list, stress_list, eq_stress)

print(elastic_tensor.property_dict)
```

What I get is:
`{'k_voigt': 9.845064083735359, 'k_reuss': 9.845063833881039, 'k_vrh': 9.845063958808199, 'g_voigt': 8.0023897488912, 'g_reuss': 4.5904549461157655, 'g_vrh': 6.2964223475034835, 'universal_anisotropy': 3.716336252207812, 'homogeneous_poisson': 0.23641618649298335, 'y_mod': 15569997014.898907}
`

This seems wrong since the bulk modulus of aluminum should be around 76 GPa, and not around 10 GPa. Can anyone point me in the correct direction, or tell me what have I missed? I would very much appreciate it.

**Comment by bowen-bd** [comment, marked as answer]
```
from chgnet.model.dynamics import EquationOfState
import numpy as np

eos = EquationOfState()
eos.fit(structure) # the structure get from mp-134 on Materials project
print(eos.get_bulk_modulus(unit='GPa'))
```

The above code gives me 62.086 GPa (or 0.3875 eV/A^3) using [CHGNet 0.3.0](https://github.com/CederGroupHub/chgnet/releases/tag/v0.3.0)

This is very similar to the bulk modulus of [Al (mp-134)](https://next-gen.materialsproject.org/materials/mp-134?material_ids=mp-134#equations_of_state) on Materials Project

**Reply by shivang-22** [reply]
Thank you @BowenD-UCB ! Would this also work for systems not in the materials project? For example, if I have aluminum but with 10% titanium in a certain atomic configuration, will CHGNet give me an accurate enough bulk modulus, or is that capability not yet known?

**Reply by bowen-bd** [reply]
CHGNet should have reasonably good extrapolation for structures not in Materials Project, the expected error is shown in the validation and test set error (which we didn't use to train the model), and also [Matbench Discovery](https://matbench-discovery.materialsproject.org/).

**Reply by shivang-22** [reply]
Okay. I see that CHGNet has an inbuilt capability to calculate the bulk modulus using EOS, but is there a capability to evaluate other similar properties like shear modulus, Young's modulus, or yield strength?

**Reply by janosh** [reply]
Have a look at [`matcalc`](https://github.com/materialsvirtuallab/matcalc/blob/3d900716155ab65078d157191963a5564a6dc233/matcalc/eos.py#L19).

**Reply by shivang-22** [reply]
Thanks for the quick reply. But from what I understand, that too calculates only the bulk modulus using the Birch-Murnaghan EOS. I'm trying to evaluate other similar mechanical properties using CHGNet, if possible.

**Reply by janosh** [reply]
[`ElasticityCalc`](https://github.com/materialsvirtuallab/matcalc/blob/3d900716155ab65078d157191963a5564a6dc233/matcalc/elasticity.py#L81) gives you the full elastic tensor.
